Engine/CMakeLists.txt

```diff
@@ -171,102 +184,175 @@ set(VENDOR_PATH ${CMAKE_CURRENT_SOURCE_DIR}/vendor)
 # include dirs
 set(ENGINE_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/src)
 
-set(ENGINE_INCLUDE_DIRS
-${ENGINE_SOURCE_DIR}
-${VENDOR_PATH}/GLAD/include
-${VENDOR_PATH}/GLFW/include
-${VENDOR_PATH}/lia/include
-${VENDOR_PATH}/spdlog/include
-${VENDOR_PATH}/stb/include
-${VENDOR_PATH}/json/include
-${VENDOR_PATH}/freetype/include
-${VENDOR_PATH}/fmt/include
+# ------------------
+# Dependencies
+# ------------------
+FetchContent_Declare(
+        glfw
+        GIT_REPOSITORY "https://github.com/glfw/glfw.git"
+        GIT_TAG "3.4"
+)
+FetchContent_MakeAvailable(glfw)
+
+FetchContent_Declare(
+        lia
+        GIT_REPOSITORY "https://github.com/denyskryvytskyi/lia.git"
+        GIT_TAG "main"
+)
+FetchContent_MakeAvailable(lia)
+
+FetchContent_Declare(
+        spdlog
+        GIT_REPOSITORY "https://github.com/gabime/spdlog.git"
+        GIT_TAG "v1.14.1"
+)
+FetchContent_MakeAvailable(spdlog)
+
+FetchContent_Declare(json URL "https://github.com/nlohmann/json/releases/download/v3.11.3/json.tar.xz")
+FetchContent_MakeAvailable(json)
+
+FetchContent_Declare(
+        freetype
+        GIT_REPOSITORY "https://github.com/freetype/freetype.git"
+        GIT_TAG "VER-2-13-2"
+)
+
+set(FT_DISABLE_ZLIB ON CACHE BOOL "" FORCE)
+set(FT_DISABLE_BZIP2 ON CACHE BOOL "" FORCE)
+set(FT_DISABLE_PNG ON CACHE BOOL "" FORCE)
+set(FT_DISABLE_HARFBUZZ ON CACHE BOOL "" FORCE)
+set(FT_DISABLE_BROTLI ON CACHE BOOL "" FORCE)
```

Comment:
These were present in your bat script, so I copied them

Reply:
Got it, works for me with the last changes, thanks.
I don't think we need to change post-build commands in this PR. It's already huge, let's avoid complexity increasing.
I'll think about how to reduce irrKlang copying in every project.